test/unit/core/type_traits/range_iterator_test.cpp

```diff
@@ -184,84 +192,88 @@ TEST(range_and_iterator, difference_type_)
-
-    using comp_list = type_list<size_t,
-                                size_t,
-                                size_t,
-                                size_t,
-                                size_t,
-                                size_t,
-                                size_t>;
+    auto v = std::views::iota(0) | seqan3::views::take_exactly(2);
+    using type_list_example = seqan3::type_list<seqan3::size_type_t<std::vector<int>>, // short
+                                                typename seqan3::size_type<std::vector<int>>::type, // long
+                                                typename std::vector<int>::size_type, // member type
+                                                seqan3::size_type_t<std::vector<int> const>, // const container
+                                                seqan3::size_type_t<std::ranges::iterator_t<std::vector<int>>>,
+                                                                                                            // iterator
+                                                seqan3::size_type_t<foreign_iterator>, // iterator2
+                                                seqan3::size_type_t<decltype(v)>>; // range, no member
+
+    using comp_list = seqan3::type_list<size_t,
+                                        size_t,
+                                        size_t,
+                                        size_t,
+                                        size_t,
+                                        size_t,
+                                        size_t>;
 
     expect_same_types<type_list_example, comp_list>();
 }
 
 TEST(range_and_iterator, innermost_value_type_)
 {
-    using type_list_example = type_list<typename innermost_value_type<std::vector<int>>::type,                    // long
-                                        innermost_value_type_t<std::vector<int>>,                                 // short
-                                        innermost_value_type_t<std::vector<std::vector<int>>>,                    // two-level
-                                        innermost_value_type_t<std::ranges::iterator_t<std::vector<int>>>,        // iterator
-                                        innermost_value_type_t<std::ranges::iterator_t<std::vector<int> const>>>; // const_iterator
-
-    using comp_list = type_list<int,
-                                int,
-                                int,
-                                int,
-                                int>;
+    using type_list_example = seqan3::type_list<typename seqan3::innermost_value_type<std::vector<int>>::type, // long
+                                                seqan3::innermost_value_type_t<std::vector<int>>, // short
+                                                seqan3::innermost_value_type_t<std::vector<std::vector<int>>>,
+                                                                                                            // two-level
```

Comment:
...

Reply:
These lines will all be too wide now, I hope that was clear to you.

Reply:
I have corrected the last commit to a hopefully nicer solution.